samples/ar-view-node/src/main/java/io/github/sceneview/sample/arviewnode/nodes/TrackingNode.kt

```diff
@@ -0,0 +1,305 @@
+            currentHittestAttempt++
+            isHittestPaused = true
+            return isHitting
+        }
+        if (!isTracking()) return false
+        if (frame == null) return false
+        if (frame.camera.trackingState != TrackingState.TRACKING) return false
+
+        return try {
+            val pt = getScreenCenter()
+            updateHitTestInternal(frame, pt)
+        } catch (e: NotTrackingException) {
+            resetValues()
+            false
+        } catch (e: DeadlineExceededException) {
+            resetValues()
+            false
+        } catch (e: FatalException) {
+            resetValues()
+            false
+        }
+    }
+
+    private fun updateHitTestInternal(
+        frame: Frame,
+        pt: android.graphics.Point
+    ): Boolean {
+        resetValues()
+
+        val hits = frame.hitTest(pt.x.toFloat(), pt.y.toFloat())
+
+        for (hit in hits) {
+            val trackable = hit.trackable
+
+            // Check what kind of Trackable we have.
+            isHitting = when (trackable) {
+                is Plane -> validatePlane(trackable, hit)
+                is DepthPoint -> validateDepthPoint(trackable)
+                is Point -> validatePoint(trackable)
+                else -> false
+            }
+
+            // If we have found a valid Trackable leave the for-loop.
+            if (isHitting) {
+                lastHitResult = hit
+                break
+            }
+        }
+        return isHitting
+    }
+
+
+    private fun getScreenCenter(): android.graphics.Point {
+        return android.graphics.Point(sceneView.width / 2, sceneView.height / 2)
+    }
+
+    private fun validatePlane(trackable: Trackable, hit: HitResult): Boolean {
+        return (trackable as Plane).isPoseInPolygon(hit.hitPose) && trackable.trackingState == TrackingState.TRACKING
+    }
+
+    private fun validatePoint(trackable: Trackable): Boolean {
+        return (trackable as Point).orientationMode === Point.OrientationMode.ESTIMATED_SURFACE_NORMAL && trackable.trackingState == TrackingState.TRACKING
+    }
+
+    private fun validateDepthPoint(trackable: Trackable): Boolean {
+        return trackable.trackingState == TrackingState.TRACKING
+    }
+
+    private fun resetValues() {
+        currentHittestAttempt = 0
+        isHitting = false
+        isHittestPaused = false
+    }
+    // endregion UpdateIsHitting
+    //////////////////
+
+
+    /////////////////////
+    // region UpdateTrackingNode
+    private fun updateTrackingNode() {
+        if (trackingNode == null &&
+            isHitting &&
+            !createCalled
+        ) {
+            createCalled = true
+            create()
+
+        } else if (trackingNode != null &&
+            createCalled
+        ) {
+            update()
+        }
+    }
+
+    private fun create() {
+        getLastHitResult()?.let { hitResult ->
+            hitResult.createAnchorOrNull().apply {
```

Comment:
I think the anchor is still created here.

Reply:
fixed